src/arm/BBORG_MOTOR-00A2.dts

```diff
@@ -9,135 +12,97 @@
 /dts-v1/;
 /plugin/;
 
-#include <dt-bindings/gpio/gpio.h>
-#include <dt-bindings/pinctrl/am33xx.h>
+/*
+ * Helper to show loaded overlays under: /proc/device-tree/chosen/overlays/
+ */
+&{/chosen} {
+	overlays {
+		BBORG_MOTOR-00A2 = __TIMESTAMP__;
+	};
+};
 
-/ {
-	/*
-	 * Helper to show loaded overlays under: /proc/device-tree/chosen/overlays/
-	 */
-	fragment@0 {
-		target-path="/";
-		__overlay__ {
+/*
+ * Free up the pins used by the cape from the pinmux helpers.
+ */
+&ocp {
+	P8_13_pinmux { status = "disabled"; };	/* motor3 speed */
+	P8_14_pinmux { status = "disabled"; };	/* motor3 direction */
+	P8_16_pinmux { status = "disabled"; };  /* motor2 direction */
+	P8_18_pinmux { status = "disabled"; };	/* motor1 direction */
+	P8_19_pinmux { status = "disabled"; };	/* motor4 speed */
+	P8_26_pinmux { status = "disabled"; };  /* motor4 direction */
+	P9_14_pinmux { status = "disabled"; };	/* motor2 speed */
+	P9_16_pinmux { status = "disabled"; };	/* motor1 speed */
+};
 
-			chosen {
-				overlays {
-					BBORG_MOTOR-00A2 = __TIMESTAMP__;
-				};
-			};
-		};
-	};
+/*
+ * Enable and Update the default state of the pins
+ */
+&ocp {
+	P8_13_pinmux { status = "okay"; pinctrl-0 = <&P8_13_pwm_pin>; };	/* motor3 speed */
```

Comment:
Given this approach, I don't think disabling the pinmux helper is needed. As only loosely mentioned in the comment, what you are doing here is changing the default mode with the pinmux helper, rather than using the driver to simply define the mode. This may be an OK approach, but it is a bit error-prone if someone messes with the pinmux helper. This change needs to be a bit better documented for users in some kind of changelog.

Reply:
Yes, you are right we don't need those disabling I will try things without it. There was a much bigger problem in the previous implementation method. we were limiting users to use labels we defined for the buses. With this approach and good documentation, the user can create new overlays without chaing in the compatibility code. I will write the documentation soon :)